sway/commands.c

```diff
@@ -169,6 +170,72 @@ static bool cmd_exit(struct sway_config *config, int argc, char **argv) {
 	return true;
 }
 
+static bool cmd_floating(struct sway_config *config, int argc, char **argv) {
+	if (strcasecmp(argv[0], "toggle") == 0) {
+		swayc_t *view = get_focused_container(&root_container);
+		// Prevent running floating commands on things like workspaces
+		if (view->type != C_VIEW) {
+			return true;
+		}
+		int i;
+		// Change from nonfloating to floating
+		if (!view->is_floating) {
+			view->is_floating = true;
+			for (i = 0; i < view->parent->children->length; i++) {
+				if (view->parent->children->items[i] == view) {
+					// Cut down on width/height so it's obvious that you've gone floating
+					// if this is the only view
+					view->width = view->width - 30;
+					view->height = view->height - 30;
```

Comment:
The i3 behaviour is to adopt whatever size the window would have had if we weren't managing it. Perhaps we can save `desired_width` and `desired_height` whenever the client makes a geometry request?